### PR TITLE
Rename the vec module to better reflect that it provides SIMD optimized vector scorers

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit renames the vector module to better reflect its intent - to provide SIMD optimized vector scorer implementations.

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
LGTM, I think all the places are covered (and CI will tell us if something is missing)

**Review by mark-vieira — COMMENTED** [review]
Do we care about what the binary artifact is named?

File: test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ static Map<String, URL> getCodebases() {
         addClassCodebase(codebases, "elasticsearch-core", "org.elasticsearch.core.Booleans");
         addClassCodebase(codebases, "elasticsearch-cli", "org.elasticsearch.cli.Command");
         addClassCodebase(codebases, "elasticsearch-preallocate", "org.elasticsearch.preallocate.Preallocate");
-        addClassCodebase(codebases, "elasticsearch-vec", "org.elasticsearch.vec.VectorScorerFactory");
+        addClassCodebase(codebases, "elasticsearch-vec", "org.elasticsearch.simdvec.VectorScorerFactory");
```
Comment (lkts) [inline test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java:226]: `elasticsearch-vec` here should be also changed to `elasticsearch-simdvec`

**Pull request merged**